src/OpenTelemetry.ResourceDetectors.Azure/AppServiceResourceDetector.cs

```diff
@@ -80,7 +80,11 @@ public Resource Detect()
         string? websiteResourceGroup = Environment.GetEnvironmentVariable(ResourceAttributeConstants.AppServiceResourceGroupEnvVar);
         string websiteOwnerName = Environment.GetEnvironmentVariable(ResourceAttributeConstants.AppServiceOwnerNameEnvVar) ?? string.Empty;
 
+#if NET6_0_OR_GREATER
+        int idx = websiteOwnerName.IndexOf('+', StringComparison.Ordinal);
+#else
         int idx = websiteOwnerName.IndexOf("+", StringComparison.Ordinal);
+#endif
```

Comment:
From current implementation:
```
public int IndexOf(char value, StringComparison comparisonType)
{
    switch (comparisonType)
    {
        case StringComparison.CurrentCulture:
        case StringComparison.CurrentCultureIgnoreCase:
            return CultureInfo.CurrentCulture.CompareInfo.IndexOf(this, value, GetCaseCompareOfComparisonCulture(comparisonType));

        case StringComparison.InvariantCulture:
        case StringComparison.InvariantCultureIgnoreCase:
            return CompareInfo.Invariant.IndexOf(this, value, GetCaseCompareOfComparisonCulture(comparisonType));

        case StringComparison.Ordinal:
            return IndexOf(value);

        case StringComparison.OrdinalIgnoreCase:
            return CompareInfo.Invariant.IndexOf(this, value, CompareOptions.OrdinalIgnoreCase);

        default:
            throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparisonType));
    }
```
It should allow remove conditional compilation

```suggestion
        int idx = websiteOwnerName.IndexOf('+');
```

Reply:
I'd like to not rely on the implementation details here. It would be fine to use `IndexOf('+')` today but .NET team is free to change the implementation in .NET 9 or later.